docs: router request lifecycle (#6497)
apollographql/router: docs/shared/router-lifecycle-services.mdx

A router's request lifecycle has three major services that support instrumentation:

> * **Router service** - Handles the opaque bytes of an incoming HTTP request before it's parsed into a GraphQL request. Operates within an HTTP server.

Author: ah, query analysis isn't explained at all across our docs, thanks for filling that gap. Updating to the following:

```suggestion
* **Router service** - Operates within the context of an HTTP server, handling the opaque bytes of an incoming HTTP request. Does query analysis to parse the GraphQL operation and validate it against schema.
```

> * **Router service** - Operates within the context of an HTTP server, handling the opaque bytes of an incoming HTTP request. Does query analysis to parse the GraphQL operation and validate it against schema.

Author: Learned from @devfreddy-apollo's review comments about the router's query analysis layer, it's a term that I'll add to apollopedia and use more throughout router docs.

> * **Supergraph service** - Handles a GraphQL request after it has been parsed and before it is sent to subgraphs. Runs the query planner to produce a query plan to execute.

Reviewer: It's probably more appropriate to say "after it has been planned," not "parsed," as that happens earlier in the Router service.

Author: ```suggestion
* **Supergraph service** - Handles a GraphQL request after it has been planned and before it is sent to subgraphs. Runs the query planner to produce a query plan to execute.
```

> * **Supergraph service** - Handles a GraphQL request after it has been planned and before it is sent to subgraphs. Runs the query planner to produce a query plan to execute.

Author: oops, the order's intentional but "planned" isn't the correct word. Should be something about after parsing/validation of the Router service 

```suggestion
* **Supergraph service** - Handles a GraphQL request after it's been parsed and validated, and before it's sent to subgraphs. Runs the query planner to produce a query plan to execute.
```
